library/src/main/java/com/chuckerteam/chucker/internal/ui/transaction/TransactionListFragment.kt

```diff
@@ -92,21 +100,42 @@ internal class TransactionListFragment :
     }
 
     override fun onOptionsItemSelected(item: MenuItem): Boolean {
-        return if (item.itemId == R.id.clear) {
-            AlertDialog.Builder(requireContext())
-                .setTitle(R.string.chucker_clear)
-                .setMessage(R.string.chucker_clear_http_confirmation)
-                .setPositiveButton(
-                    R.string.chucker_clear
-                ) { _, _ ->
-                    RepositoryProvider.transaction().deleteAllTransactions()
-                    NotificationHelper.clearBuffer()
+        return when (item.itemId) {
+            R.id.clear -> {
+                AlertDialog.Builder(requireContext())
+                    .setTitle(R.string.chucker_clear)
+                    .setMessage(R.string.chucker_clear_http_confirmation)
+                    .setPositiveButton(
+                        R.string.chucker_clear
+                    ) { _, _ ->
+                        RepositoryProvider.transaction().deleteAllTransactions()
+                        NotificationHelper.clearBuffer()
+                    }
+                    .setNegativeButton(R.string.chucker_cancel, null)
+                    .show()
+                true
+            }
+            R.id.export -> {
+                if (transactions.isEmpty()) {
+                    view?.let { Snackbar.make(it, R.string.chucker_export_empty_text, Snackbar.LENGTH_SHORT).show() }
+                } else {
+                    AlertDialog.Builder(requireContext())
+                        .setTitle(R.string.chucker_export)
+                        .setMessage(R.string.chucker_export_http_confirmation)
+                        .setPositiveButton(R.string.chucker_export) { _, _ ->
+                            val exportable = transactions.joinToString(
+                                separator = "\n${getString(R.string.chucker_export_separator)}\n",
+                                prefix = "${getString(R.string.chucker_export_prefix)}\n",
+                                postfix = "\n${getString(R.string.chucker_export_postfix)}\n"
+                            ) { "\n${FormatUtils.getShareText(context, it)}\n" }
+                            requireContext().share(exportable, "chucker_export", "chucker_export")
```

Comment:
Please extra this logic into a separate class (e.g. Util class) or method. A Junit test for this would also be a plus :+1: